python/ql/src/experimental/Security/CWE-352/DjangoCSRF.ql

```diff
@@ -0,0 +1,36 @@
+/**
+ * @name Disabled Django CSRF protection
+ * @description No csrf Middleware/ Decorator to protect Django application from CSRF
+ * @kind problem
+ * @problem.severity error
+ * @precision high
+ * @id py/django-csrf
+ * @tags security
+ *       external/cwe/cwe-352
+ */
+
+import python
+import semmle.python.strings
+
+from GlobalVariable g
+where
+  /*
+   * Regex matching MIDDLEWARE & MIDDLEWARE_CLASSES(backwards compatibility)
+   * Checks if there is a GlobalVariable(List/Tuple) named MIDDLEWARE or MIDDLEWARE_CLASES
+   * and contains the middleware string
+   */
+
+  g.getId().regexpMatch("MIDDLEWARE(_CLASSES)?") and
```

Comment:
```suggestion
  g.getId() in ["MIDDLEWARE", "MIDDLEWARE_CLASSES"] and
```
This is more efficient in CodeQL. Think of the QL evaluator as a database engine. Using direct lookup like this will allow it to use an index, whereas regexp matching forces it to look at all rows.